docs/architecture/cataloging/README.md

```diff
@@ -0,0 +1,9 @@
+# Cataloging
```

Comment:
Should we rename this (and the containing folder) to "Catalog"? all the other folders at this same level have noun-style naming

Reply:
I thought of "the cataloging" as term, but I can change it